src/utils/S3Utils.ts

```diff
@@ -8,12 +8,14 @@ import {
   PutObjectCommand,
   S3Client
 } from '@aws-sdk/client-s3';
+import { v4 as uuid } from 'uuid';
 import { S3Object } from '../types/S3Object';
 
 const fileToS3Object = (path: string, object: any): S3Object => {
   const name = object.Key.endsWith('/') ? object.Key.split('/').slice(-2, -1)[0] : object.Key.split('/').pop();
 
   return {
+    id: object.Metadata?.id ? object.Metadata.id : '',
```

Comment:
Might want to tweak this logic a little. Could potentially update the metadata if the ID is missing. Just thinking of how to handle the files that are already uploaded.